wolfcrypt/src/pkcs7.c

```diff
@@ -11868,8 +11872,23 @@ WOLFSSL_API int wc_PKCS7_DecodeAuthEnvelopedData(PKCS7* pkcs7, byte* in,
             pkiMsgSz = (pkcs7->stream->length > 0)? pkcs7->stream->length: inSz;
 
             encryptedContentSz = pkcs7->stream->expected;
+        #else
+            pkiMsgSz = inSz;
         #endif
 
+            if (expBlockSz == 0) {
+                if (GetAlgoId(pkiMsg, &idx, &encOID, oidBlkType, pkiMsgSz) < 0)
```

Comment:
This is close but not quite right I think for streaming cases. If the function has exited out this GetAlgoId will be called with an advanced 'idx' value that will read into a location in the input buffer that might not have the AlgoID. Since encOID has been stored in the PKCS7 struct earlier for streaming cases we can skip the GetAlgoId function call and just do wc_PKCS7_GetOIDBlockSize() with the stored encOID value.